web/src/user_group_edit.ts

```diff
@@ -1539,21 +1539,66 @@ export function update_empty_left_panel_message(): void {
     let has_groups;
     const is_your_groups_tab_active =
         get_active_data().$tabs.first().attr("data-tab-key") === "your-groups";
+
+    const current_group_filter = filters_dropdown_widget.value();
+    const include_deactivated_groups =
+        current_group_filter === FILTERS.DEACTIVATED_GROUPS ||
+        current_group_filter === FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
+
+    const groups = user_groups.get_user_groups_of_user(
+        people.my_current_user_id(),
+        include_deactivated_groups,
+    );
+
     if (is_your_groups_tab_active) {
-        has_groups = user_groups.get_user_groups_of_user(people.my_current_user_id(), true).length;
+        has_groups =
+            current_group_filter === FILTERS.DEACTIVATED_GROUPS
+                ? groups.filter((group) => group.deactivated).length
+                : groups.length;
     } else {
-        has_groups = user_groups.get_realm_user_groups().length;
+        has_groups =
+            current_group_filter === FILTERS.DEACTIVATED_GROUPS
+                ? user_groups.get_realm_user_groups(true).filter((group) => group.deactivated)
+                      .length
+                : user_groups.get_realm_user_groups(include_deactivated_groups).length;
     }
```

Comment:
I don't understand the refactoring here, but it's very hard to read. Is there a reason this is better than the logic we had before? 

In any case, please avoid complex logic in `?` comprehensions split across lines this way; it's not very readable.